lib/services/auth-service.js

```diff
@@ -0,0 +1,149 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+var jwt = require('jsonwebtoken');
+var crypto = require('crypto');
+
+module.exports = authServiceFactory;
+di.annotate(authServiceFactory, new di.Provide('Auth.Services'));
+di.annotate(authServiceFactory,
+    new di.Inject(
+        'Assert',
+        'Services.Configuration'
+    )
+);
+
+function authServiceFactory(
```

Comment:
Nice suggestion @brianparry. 

Bcrypt is not bad either. Both of them can meets our needs. We did compare bcrypt and crypto.pbkdf2 before we start coding and crypo.pbkdf2 was the choice at that time. 

The discussion is listed in google group discussion at https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/rackhd/Hw88shMffLw